### PR TITLE
SW-2530 Create generic tooltip component

### PR DESCRIPTION
**Pull request by alexTerraformation**
To be added to icon buttons.

For the future we could consider replacing the IconTooltip with this generic version but I didn't do that here.

### CODE REVIEW

**Review by constanzauanini — APPROVED** [review]
Lgtm! Yes, we should re-use it in icon tooltip later because they have the same code/styles

**Review by charlie-tf — COMMENTED** [review]
Looks good! The hover interaction is a bit unexpected. It makes sense on non-mobile, but for mobile the tooltip appears on click and closes when the mouse cursor is moved away. Not sure how it behaves on an _actual_ mobile with touch though

**Comment by charlie-tf** [comment]
> Looks good! The hover interaction is a bit unexpected. It makes sense on non-mobile, but for mobile the tooltip appears on click and closes when the mouse cursor is moved away. Not sure how it behaves on an _actual_ mobile with touch though

Okay, I tried it on my phone. The tooltip appears on tapping the button, as expected, and disappears when defocusing, also as expected. It goes away on its own after a couple seconds without any interaction as well.

**Review by charlie-tf — APPROVED** [review]
Nice clean component code.

**Comment by alexTerraformation** [comment]
> > Looks good! The hover interaction is a bit unexpected. It makes sense on non-mobile, but for mobile the tooltip appears on click and closes when the mouse cursor is moved away. Not sure how it behaves on an _actual_ mobile with touch though
> 
> Okay, I tried it on my phone. The tooltip appears on tapping the button, as expected, and disappears when defocusing, also as expected. It goes away on its own after a couple seconds without any interaction as well. This seems fine 👍

This comment got me thinking a bit and I realized my solution (and the icontooltip) would not handle responsive transitions because it would try to go controlled<->uncontrolled with the component.  You can demo this in storybook with the icontooltip currently.  I updated it so that it is now always controlled.  If this solution works well then I can port to the icontooltip or replace the icontooltip with this one.

File: src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  const [open, setOpen] = React.useState(false);
+  const { isMobile } = useDeviceInfo();
+  const classes = useStyles();
+
+  const handleTooltipClose = () => {
+    setOpen(false);
+  };
+
+  const handleTooltipOpen = (event: React.SyntheticEvent | Event) => {
+    if(event.type === 'click' || (event.type === 'mouseover' && !isMobile)){
```
Comment (charlie-tf) [inline src/components/Tooltip/Tooltip.tsx:31]: 🎉

**Review by charlie-tf — APPROVED** [review]
New commits look good to me. And it works well in vercel

**Pull request merged**